searx/utils.py

```diff
@@ -137,6 +138,11 @@ def handle_entityref(self, name):
     def get_text(self):
         return ''.join(self.result).strip()
 
+    def error(self, message):
+        # error handle is needed in <py3.10
+        # https://github.com/python/cpython/pull/8562/files
+        raise AssertionError(message)
```

Comment:
The tests might not work because of the usage of AssertionError here. A dedicated exception would be better.

Reply:
Understand .. but AssertionError is what >=py3.10 raises / see code from cpython [implementation](https://github.com/python/cpython/pull/8562/files#diff-7304624ffcc8be1268e5f86ebf1723034477694ce33456dc330f9c03f2d1e84fL134-R132).

BTW: I think the problem in tests is not the raise of AssertionError .. the problem might be that we need an Exception handling of AssertionError (see `except AssertionError` block in line 171)

Reply:
@dalf After thinking about it again I realized that a ``AssertionError``exception is handled in the test framework like any other exception. But I took the discussion as a reason to add a test for the problematic cases --> [diff](https://github.com/searxng/searxng/compare/01cfc55b5f595c9d5a265646d09035cd934e24f6..89039104a481bde12530b6494986ff71727b5d60)